### PR TITLE
Use SDK response in playground

### PR DESCRIPTION
**Pull request by andresgutgon**
# What?
Use SDK to fetch data from gateway and use it in Playground

## TODO
- [x] SDK accumulates conversation messages and return at the end
- [x] Add tests for `runDocument`
- [x] Handle errors for example wrong model name

### CODE REVIEW

File: packages/core/src/services/commits/runDocumentAtCommit.ts [hunk 1]
```diff
@@ -179,10 +180,29 @@ async function iterate({
...
+    controller.close()
+    return {
+      text: error.message,
+      usage: {
+        promptTokens: 0,
+        completionTokens: 0,
+        totalTokens: 0,
+      },
+      toolCalls: [],
+    }
```
Comment (andresgutgon) [inline packages/core/src/services/commits/runDocumentAtCommit.ts:205]: This way we catch the error from AI apis providers
image

File: packages/sdks/typescript/src/test/chunks-example.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+          promptTokens: 114,
+          completionTokens: 352,
+          totalTokens: 466,
+        },
+      },
+    },
+  },
+]
+
+export const FINAL_RESPONSE = {
```
Comment (andresgutgon) [inline packages/sdks/typescript/src/test/chunks-example.ts:153]: @csansoon this is the output of `runDocument`

File: apps/web/src/actions/sdk/runDocumentAction.ts [hunk 3]
```diff
@@ -22,26 +22,30 @@ export async function runDocumentAction({
...
+          name: error.name,
+          message: error.message,
+          stack: error.stack,
+        })
+      },
+      onFinished: () => stream.done(),
+    })
+  }
+
+  runSdk()
```
Comment (csansoon) [inline apps/web/src/actions/sdk/runDocumentAction.ts:48]: Why is this defined as a function? Can't you just do it directly?

**Review by csansoon — DISMISSED** [review]

File: apps/web/src/actions/sdk/runDocumentAction.ts (hunk 3 above)
Comment (andresgutgon) [reply apps/web/src/actions/sdk/runDocumentAction.ts]: Removed

File: packages/sdks/typescript/src/index.ts [hunk 4]
```diff
@@ -1,12 +1,24 @@
...
 import { RouteResolver } from '$sdk/utils'
-import {
-  ChainEvent,
-  ChainEventTypes,
-  HandlerType,
-  StreamEventTypes,
-} from '$sdk/utils/types'
+import { ChainEvent, HandlerType } from '$sdk/utils/types'
 
+type Message = {
```
Comment (andresgutgon) [inline packages/sdks/typescript/src/index.ts:5]: My OCD is going to explode. We need to DRY all these types

**Pull request merged**